metapool/calculate_coefficients.py

```diff
@@ -0,0 +1,147 @@
+import scipy
+import seaborn as sns
+import pandas as pd
+import numpy as np
+
+
+def plot_with_fit(xcol, ycol, groups, xmax=4, ymax=4,
+                  textpos=(.6, .2)):
+    from matplotlib import pyplot as plt
+    import matplotlib
+    matplotlib.use("TkAgg")
+    x = 0
+    y = 0
+    fig, axes = plt.subplots(xmax, ymax, sharex=True, sharey=True)
+    for name, group in groups:
+        ax = axes[y, x]
+        # ax.scatter(group["CPMlog"], group["Dilution2"])
+
+        # TODO FIXME HACK: Unclear what R^2 value is being printed by ggplot
+        #  but it definitely isn't a pearson computed per group.  *shrug*
+        no_nan = group.dropna()
+        r, p = scipy.stats.pearsonr(no_nan[xcol], no_nan[ycol])
+        sns.regplot(x=xcol, y=ycol, data=group, ax=ax)
+        ax.text(textpos[0], textpos[1], 'r={:.2f}\np={:.2g}'.format(r, p),
+                transform=ax.transAxes)
+        if x > 0:
+            ax.set_ylabel("")
+        if y < ymax - 1:
+            ax.set_xlabel("")
+
+        ax.set_title("\n".join(name))
+        x += 1
+        if x == xmax:
+            x = 0
+            y += 1
+        if y == ymax:
+            plt.show()
+            x = 0
+            y = 0
+
+    if x != 0 or y != 0:
+        plt.show()
+
+
+def calculate_coefficients(table_synthetic_hits, metadata_pools, dilutions,
+                           plot_fit=False):
+    # Strip leading X in sample names which are automatically removed by R
+    # table reading.
+    metadata_pools["sample_name_r"] = metadata_pools["sample_name_r"].str[1:]
```

Comment:
Can you use sample_name vs. sample_name_r? Also, could you make sure this is the case for all the code in this PR? Thank you.

Reply:
Done